README.md

```diff
@@ -1,12 +1,60 @@
-# How to use these scripts
-* ./benchmakrk.py --measure ./results
-* ./benchmakrk.py --profile ./profile
+# Setting Up
+## Python Virtual Environment
+This project uses python [Virtual Environments](https://docs.python.org/3/tutorial/venv.html).
+
+Create the venv. Only needs to be done once.
+```
+python3 -m venv ./env
+```
+Activate the venv.
+```
+source ./env/bin/activate
+```
+Once the venv is activated, `python` refers to python3.
+Upgrade pip and install all python dependancies. 
+```
+python -m pip install -U pip
+python -m pip install -r requirements.txt
+```
+
+To deactivate venv.
+```
+deactivate
+```
+
+## Getting around sudo password
+To disable asking password for sudo commands (required by powermetrics).
+Run `sudo visudo` and add the last line to User specification (replacing `<user>`):
+```
+# root and users in group wheel can run anything on any machine as any user
+root ALL = (ALL) ALL
+%admin ALL = (ALL) ALL
+<user> ALL = (ALL) NOPASSWD:ALL
+```
+
+## dtrace
+Running benchmark.py in profile mode uses `dtrace` to analyse the chromium processes. By default `dtrace` does not work well with [SIP](https://support.apple.com/en-us/HT204899). Disabling SIP as a whole is not recommended and instead should be done only for dtrace using these steps:
+
+* Reboot in recovery mode
+* Start a shell
+* Execute `csrutil enable --without dtrace --without debug`
+* Reboot
+
+=======
+=======
 
 ## benchmark.py
 Use to execute different usage scenarios and measure their power use using powermetrics.
+```
+./benchmakrk.py --measure ./results
+./benchmark.py profile --profile_mode cpu_time --chromium_executable=./bin/Chromium.app
```

Comment:
Is "profile" the output dir? Let's put "./profile" for consistency

Reply:
Sure! I fixed some problems with the other command too.